dds/DCPS/TypeAssignability.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#ifndef OPENDDS_DCPS_TYPE_ASSIGNABILITY_H
+#define OPENDDS_DCPS_TYPE_ASSIGNABILITY_H
+
+#include "dds/DCPS/TypeObject.h"
```

Comment:
```suggestion
#include "TypeObject.h"
```

Reply:
I saw some headers are included using the `dds/DCPS/` prefix, and some are not. Do we have a rule for when to use which?

Reply:
For this specifically, no, but it probably should be, because I don't see a reason why to use a include like this for a file that's in the same directory. [These are the rules on includes we have right now](https://github.com/objectcomputing/OpenDDS/blob/master/docs/guidelines.md#includes).

Reply:
Right, I think it falls into the general rule: if you can express the same concept in fewer lines/characters and it's still clear and maintainable, then that shorter form is preferred.

Reply:
I have went ahead and proposed this as a guideline: #1704

Reply:
Great. I've removed the prefix. I think we may want to update other files to use this guideline for consistency.